### PR TITLE
Add touch icons for Chrome, iOS and Windows 10

### LINKED ISSUE #112: Add a Web App Manifest so I can add to my homescreen
Site works well on mobile, it would be nice to have it a bit more stand-alone and something I can install to my homescreen.
https://developer.mozilla.org/en-US/docs/Web/Manifest

A basic Service Worker to allow some simple caching would be neat too.

### PR DESCRIPTION
**Pull request by fhemberger**
Fixes #112

### CODE REVIEW

**Comment by fhemberger** [comment]
Fixes #112.

File: public/manifest.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+	"name": "Mastodon",
```
Comment (rowan-m) [inline public/manifest.json]: I would add `short_name` and `description`.

File: public/manifest.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+	"icons": [
+		{
+			"src": "\/assets\/android-chrome-192x192.png",
+			"sizes": "192x192",
+			"type": "image\/png"
+		}
+	],
+	"theme_color": "#ffffff",
+	"display": "standalone"
```
Comment (rowan-m) [inline public/manifest.json]: I would add `"start_url": "/#/statuses/home"`

**Review by rowan-m — CHANGES_REQUESTED** [review]

File: public/manifest.json (hunk 1 above)
Comment (fhemberger) [inline public/manifest.json]: If no `short_name` is given, `name` will be used instead.

Comment (fhemberger) [inline public/manifest.json]: Added `description` and `start_url`.

**Review by rowan-m — APPROVED** [review]

File: app/views/layouts/application.html.haml [hunk 3]
```diff
@@ -5,6 +5,11 @@
     %meta{:name => 'viewport', :content => 'width=device-width, initial-scale=1'}/
     %meta{'http-equiv' => 'X-UA-Compatible', :content => 'IE=edge'}/
 
+    %link{:rel => "apple-touch-icon", :sizes => "180x180", :href => "/assets/apple-touch-icon.png"}/
+    %link{:rel => "manifest", :href => "/assets/manifest.json"}/
```
Comment (rowan-m) [inline app/views/layouts/application.html.haml]: This is currently deployed to `/manifest.json`. Either should point to the correct location or the file should be moved. Not sure what's more correct for the architecture.

Comment (fhemberger) [inline app/views/layouts/application.html.haml]: Oh, yes. You're right. @Gargron any preferences on where to put those two files?

Comment (Gargron) [inline app/views/layouts/application.html.haml]: Right... Actually, manifest.json links to some assets, but you have to be aware that they're compiled in the asset pipeline and the end result will have slightly different file names (with digests) which you can't know in advance. So manifest.json should probably be in the assets folder and be something like `manifest.json.erb` and link to the graphics using the `asset_path` helper.

**Pull request merged**